enumerables.rb

```diff
@@ -0,0 +1,127 @@
+# rubocop: disable Metrics/ModuleLength
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    n = 0
+    while n <= size - 1
+      yield(to_a[n])
+      n += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    n = 0
+    while n <= size - 1
+      yield(to_a[n], n)
+      n += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    result = []
+    my_each do |n|
+      result.push(n) if yield(n)
+    end
+    result
+  end
+
+  def my_all?(var = nil)
+    result = true
+    my_each do |n|
+      if block_given?
+        result = false unless yield(n)
+      elsif var.nil?
+        result = false unless n
+      else
+        result = false unless var == n
+      end
+    end
+    result
+  end
+
+  def my_any?(var = nil)
+    result = false
+    my_each do |n|
+      if block_given?
+        return result = true if yield(n)
+      elsif var.nil?
+        return result = true unless n
+      else
+        return result = true unless var == n
+      end
+    end
+    result
+  end
```

Comment:
* [ ] should have working `my_any?` **(covered by automated tests)** identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
  ```ruby
    array = [nil, false, true, []]
    p array.my_any? == array.any? # false
  ```

* [ ] my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
  ```ruby
    array = %w[dog door rod blade]
    p array.my_any?(Integer) == array.any?(Integer) # false
  ```
* [ ] my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
  ```ruby
    array = %w[dog door rod blade]
    p array.my_any?(/c/) == array.any?(/c/) # false
  ```
* [ ] my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
  ```ruby
    array = %w[dog door rod blade]
    p array.my_any?('cat') == array.any?('cat') # false
  ```

Reply:
Same as above, but changed the "unless" statements to "if", which is now working the way It should be.